code/game/gamemodes/nuclear/nuclearbomb.dm

```diff
@@ -8,71 +8,250 @@ var/bomb_set
 	density = 1
 
 	var/timeleft = 60.0
-	var/timing = 0.0
+	var/timing = 0
 	var/r_code = "ADMIN"
 	var/code = ""
-	var/yes_code = 0.0
-	var/safety = 1.0
+	var/yes_code = 0
+	var/safety = 1
 	var/obj/item/weapon/disk/nuclear/auth = null
 	use_power = 0
 	var/previous_level = ""
 	var/lastentered = ""
 	var/immobile = 0 //Not all nukes should be moved
+	var/obj/item/nuke_core/core = null
+	var/construction_state = 6
 
 /obj/machinery/nuclearbomb/selfdestruct
 	name = "station self-destruct terminal"
 	desc = "For when it all gets too much to bear. Do not taunt."
 	icon = 'icons/obj/machines/bignuke.dmi'
+	icon_state = "nuclearbomb_base"
 	anchored = 1 //stops it being moved
 	immobile = 1 //prevents it from ever being moved
 	layer = 4
+	var/icon/lights = null
+	var/icon/panel = null
+	var/icon/interior = null
+	var/icon/glow = null
+
+/obj/machinery/nuclearbomb/selfdestruct/New()
+	core = new /obj/item/nuke_core(src)
+	lights = new /icon(icon,"lights-off")
+	panel = new /icon(icon,"panel-overlay")
+	overlays += lights
+	overlays += panel
+	..()
+
+/obj/machinery/nuclearbomb/selfdestruct/attackby(obj/item/I, mob/user, params)
+	switch(construction_state)
+		if(6)

```

Comment:
why do this when we could just standardize it with machine deconstruction, like every other machine: screwdriver -> crowbar

(upgrade the nuke with the RPED :^)